compiler/backend/src/org/jetbrains/kotlin/codegen/inline/MethodInliner.kt

```diff
@@ -372,7 +372,7 @@ class MethodInliner(
             private fun getNewIndex(`var`: Int): Int {
                 val lambdaInfo = inliningContext.lambdaInfo
                 if (reorderIrLambdaParameters && lambdaInfo is IrExpressionLambda) {
-                    val extensionSize = if (lambdaInfo.isExtensionLambda) lambdaInfo.invokeMethod.argumentTypes[0].size else 0
+                    val extensionSize = lambdaInfo.invokeMethod.argumentTypes.take(lambdaInfo.capturedVarsStart).sumOf { it.size }
                     return when {
                         //                v-- extensionSize     v-- argsSizeOnStack
```

Comment:
Looks like scheme below should be updated

Reply:
The scheme is still correct, though I guess it isn't *guaranteed* that the thing to the left of captures is a (possibly-represented-as-value-parameter) extension receiver.